li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaConsumerImpl.java

```diff
@@ -234,8 +234,7 @@ public void unsubscribe() {
     _kafkaConsumer.unsubscribe();
   }
 
-  @Override
-  public ConsumerRecords<K, V> poll(long timeout) {
+  private ConsumerRecords<K, V> poll(long timeout, boolean includeMetadataInTimeout) {
```

Comment:
I was thinking making this API public. If we use this impl directly (e.g., kafka-rest), we can be more explicitly whether we should include the metatadata in timeout or not.  On the other head, this function is neither a public function in LikafkaConsumer interface nor in KafkaConsumer, it makes sense to be a private function.

Reply:
I'm inclined to keep the interface of `LiKafkaConsumerImpl` consistent and do not expose `poll(long, boolean)`

Reply:
I guess if we want to keep the API pure, having this private method makes sense. This means we can avoid diverging from `KafkaConsumer` interface.

The bigger problem will be if and when `poll(long timeout)` is removed we'll be at the same junction as the old scala client. We need to document exactly why `poll(long timeout)` is needed second we need to push for a new API for `KafkaConsumer`